benchmark/benchmark-atomic.cpp

```diff
@@ -0,0 +1,220 @@
+// default all atomic implementations to a desul backend.
+#include "RAJA/policy/openmp/atomic.hpp"
+#include "RAJA/RAJA.hpp"
+#include "RAJA/policy/openmp/policy.hpp"
+
+#include <type_traits>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+
+/// Conditional compilation for CUDA benchmarks.
+#if defined RAJA_ENABLE_CUDA
+#include "RAJA/policy/cuda.hpp"
+#include "RAJA/policy/cuda/atomic.hpp"
+
+template<int BLOCK_SZ>
+struct ExecPolicyGPU {
+    using policy = RAJA::cuda_exec<BLOCK_SZ, false>;
+    static std::string PolicyName() {
+        std::stringstream ss;
+        ss << "CUDA execution with block size " << BLOCK_SZ;
+        return ss.str();
+    }
+};
+
+struct GPUAtomic {
+    using policy = RAJA::policy::cuda::cuda_atomic;
+};
+
+#elif defined RAJA_ENABLE_HIP
+#include "RAJA/policy/hip.hpp"
+#include "RAJA/policy/hip/atomic.hpp"
+
+template<int M>
+struct IsGPU<RAJA::hip_exec<M>> : public std::true_type {};
+
+template<int BLOCK_SZ>
+struct ExecPolicyGPU {
+    using policy = RAJA::hip_exec<BLOCK_SZ, false>;
+    static std::string PolicyName() {
+        std::stringstream ss;
+        ss << "HIP execution with block size " << BLOCK_SZ;
+        return ss.str();
+    }
+};
+
+struct GPUAtomic {
+    using policy = RAJA::policy::hip::hip_atomic;
+};
+
+#endif
+
+#include "desul/atomics.hpp"
+#include "RAJA/util/Timer.hpp"
+
+#define BLOCK_SZ 256
+#define INDENT "  "
+using raja_default_desul_order = desul::MemoryOrderRelaxed;
+using raja_default_desul_scope = desul::MemoryScopeDevice;
+
+// Desul atomics have a different signature than RAJA's built in ops. The following code provides some
+// helper function templates so that they can be called using the same signature in timing code.
+
+// Struct holding Desul atomic signature typedef
+template<typename AtomicType>
+struct DesulAtomicSignature {
+    using type = AtomicType(*)(AtomicType*, const AtomicType, raja_default_desul_order, raja_default_desul_scope);
+};
+
+// Struct holding RAJA atomic signature typedef
+template<typename AtomicType>
+struct RajaAtomicSignature {
+    using type = AtomicType(*)(AtomicType*, const AtomicType);
+};
+
+/// RAJA::atomicAdd is overloaded and has an ambiguous type so it can't be passed as a template parameter.
+/// The following wrappers disambiguate the call and provide a signature comaptible with the DESUL
+/// wrapper.
+template<typename Policy>
+struct AtomicAdd {
+    template<typename AtomicType>
+    RAJA_HOST_DEVICE AtomicType operator()(AtomicType* acc, const AtomicType val) const {
+        return RAJA::atomicAdd(Policy {}, acc, val);
+    }
+};
+
+template<typename Policy>
+struct AtomicMax {
+    template<typename AtomicType>
+    RAJA_HOST_DEVICE AtomicType operator()(AtomicType* acc, const AtomicType val) const {
+        return RAJA::atomicMax(Policy {}, acc, val);
+    }
+};
+
+/// ExecPolicy wrapper for OpenMP
+struct ExecPolicyOMP {
+    using policy = RAJA::omp_for_exec;
```

Comment:
```suggestion
    using policy = RAJA::omp_parallel_for_exec;
```
This policy isn't going to work. Just like "#pragma omp for", it needs to be contained in a parallel region. Alternatively, you could do ``RAJA::omp_parallel_exec<RAJA::omp_for_exec>``, but it's more verbose.